### PR TITLE
feat(classification): configurable minimum values threshold

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/docs/dev_guides/classification.md [hunk 1]
```diff
@@ -6,43 +6,60 @@ The classification feature enables sources to be configured to automatically pre
 
 Note that a `.` is used to denote nested fields in the YAML recipe.
 
-| Field | Required | Type | Description | Default |
```
Comment (mayurinehate) [inline metadata-ingestion/docs/dev_guides/classification.md:9]: Formatting changes only.

File: metadata-ingestion/docs/dev_guides/classification.md [hunk 2]
```diff
@@ -6,43 +6,60 @@ The classification feature enables sources to be configured to automatically pre
...
+| confidence_level_threshold                             |                                                       | number                                         |                                                                                                                                                           | 0.68                                                                                                                                                                  |
+| info_types                                             |                                                       | list[string]                                   | List of infotypes to be predicted. By default, all supported infotypes are considered, along with any custom infotypes configured in `info_types_config`. | None                                                                                                                                                                  |
```
Comment (mayurinehate) [inline metadata-ingestion/docs/dev_guides/classification.md:33]: updated descriprion for `info_types` input.

File: metadata-ingestion/docs/dev_guides/classification.md [hunk 3]
```diff
@@ -6,43 +6,60 @@ The classification feature enables sources to be configured to automatically pre
...
+| info_types_config.`key`.values.library                 |                                                       | Array of string                                | Library used for prediction                                                                                                                               | None                                                                                                                                                                  |
+| minimum_values_threshold                               |                                                       | number                                         | Minimum number of non-null column values required to process `values` prediction factor.                                                                  | 50                                                                                                                                                                    |
```
Comment (mayurinehate) [inline metadata-ingestion/docs/dev_guides/classification.md:46]: Added new config param minimum_values_threshold and list of supported info types below.

File: metadata-ingestion/tests/integration/snowflake/test_snowflake_legacy_lineage.py [hunk 4]
```diff
@@ -50,14 +50,15 @@ def test_snowflake_basic(pytestconfig, tmp_path, mock_time, mock_datahub_graph):
...
-                "col_3": [random_cloud_region() for i in range(1, 100)],
+                "col_1": [random.randint(0, 100) for i in range(20)],
+                "col_2": [random_email() for i in range(20)],
+                "col_3": [random_cloud_region() for i in range(20)],
             }
         )
 
         datahub_classifier_config = DataHubClassifierConfig()
         datahub_classifier_config.confidence_level_threshold = 0.58
+        datahub_classifier_config.minimum_values_threshold = 10
```
Comment (hsheth2) [inline metadata-ingestion/tests/integration/snowflake/test_snowflake_legacy_lineage.py:61]: These should definitely be in the constructor and not separate statements, but can do that in a follow up

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**